### ISSUE TITLE
Can the Leffa method be migrated to Flux?

### ISSUE THREAD

**Issue by ZhengMengbin** [franciszzj/Leffa#37]
If so, at which layers should the loss be done?
I tried it initially, but it didn't work whether I added it to FluxTransformerBlock or FluxSingleTransformerBlock.

**Comment by ZhengMengbin** [comment]
> it would need an inpainting model

I use [Flux-Fill](https://huggingface.co/spaces/black-forest-labs/FLUX.1-Fill-dev), it is a inpainting model.

**Comment by ZhengMengbin** [comment]
@joekendal

**Comment by ZhengMengbin** [comment]
> because they haven't released the training code, you have just guessed it right?

Yes, the sd3 training code I referred to has no problem in outputting the image. There is no problem in adding Leffa when training, but it just doesn't work.

**Comment by ZhengMengbin** [comment]
> [#4 (comment)](https://github.com/franciszzj/Leffa/issues/4#issuecomment-2548188469)
> 
> without the loss function you can't reproduce easily

It seems so, looking forward to open source loss.

**Comment by franciszzj** [comment]
I'm very glad to see that you are trying out Flux.1 Fill, as it will lead to better performance.

Looking at your results, I think there might be some training bugs. Based on the paper, even without using Leffa loss, the results should still be comparable to previous methods.

**Comment by ruanruano** [comment]
> > because they haven't released the training code, you have just guessed it right?
> 
> Yes, the sd3 training code I referred to has no problem in outputting the image. There is no problem in adding Leffa when training, but it just doesn't work.

Hello, can I see the training code you wrote? I am also trying to write the training code but I have been having problems.

**Comment by Kuekua** [comment]
> [@joekendal](https://github.com/joekendal) After grid sample, my picture looks like this, I can't see the clothes at all ![image](https://github.com/user-attachments/assets/3f4b8d6b-303b-4c8b-ae37-5a970048be2f)

I meet the same result. Do you fix it ?
